src/main/java/teammates/ui/webapi/action/GetCourseJoinStatusAction.java

```diff
@@ -49,6 +50,13 @@ private JsonResult getInstructorJoinStatus(String regkey) {
         if (instructor == null) {
             return new JsonResult("No instructor with given registration key: " + regkey, HttpStatus.SC_NOT_FOUND);
         }
+
+        String institute = getRequestParamValue(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
```

Comment:
Why we need to verify MAC for GET action?

Reply:
If we don't then everything seems normal to the user up till the point in which he presses "Join course", then he gets thrown an "Internal Server error" message and is told to write a report to the teammates dev team. IMO it would be better to show that the MAC was invalid at the point in which the "Join course" button is available and thus let the user know that they should re-check the URL.

Reply:
> If we don't then everything seems normal to the user up till the point in which he presses "Join course"

In the normal workflow, I don't think user will encounter the situation unless he intends to modify the URL.

> he gets thrown an "Internal Server error" message

The problem looks like is in the frontend where we should handle the situation better. e.g. more helpful message like "do you copy the full URL in the join email?"

The MAC makes sure data integrity is not compromised when **creating** the account. Unlike `joinKey`, it has nothing to do with the GET operation and that is why I put the comment.

Reply:
I agree it's not great in terms of RESTful API convention - GET should only be retrieving data and not verifying that checksums match. 

Sure, we can roll back the change (so the error only appears after the user clicks on "Join Course".) The only issues I see is that (1) the user actually needs to click on Join Course (I would prefer if the error message is shown before the user has to click the button), though I suppose this may not be worth breaking RESTful convention and (2) if we replace the error message with say, a toast, then there isn't a form available to report the issue to the dev team. I'd prefer to keep that form, since it gives them an avenue to report if it really ends up being a problem within teammates.

Reply:
I'm surmising from the discussion that we'll be adopting the following workflow:

1. User navigates to join link
2. Front-end validation for presence of institute signature
2a. If signature missing, display front-end error message
3. User clicks on "Register for course"
4. Back-end validation of signature
4a. If signature wrong, display internal server error message

Error message in (2a) to display a message to the instructor to request for a new join link, as the only non-malicious use-case would be instructors who have received their join link before this feature was implemented.

Internal server error message in (4a) is the `ErrorReportComponent` modal to allow for error reporting, considering that in the non-malicious use-case, this should not happen.

Do let me know if there are any other considerations I might have missed. Thank you!

Reply:
Looks good